UnitTests/MockCaptureFixture.cs

```diff
@@ -0,0 +1,163 @@
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            
+            var expectedValues = new List<string> { "Hello!" };
+            Assert.Equal(expectedValues, capture.All);
+        }
+
+        [Fact]
+        public void CanCaptureMultipleParameters()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            var expectedValues = new List<string> { "Hello!", "World!" };
+            Assert.Equal(expectedValues, capture.All);
+        }
+
+        [Fact]
+        public void CanCaptureSpecificParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.Capture(p => p.StartsWith("W"))));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            var expectedValues = new List<string> { "World!" };
+            Assert.Equal(expectedValues, capture.All);
+        }
+
+        [Fact]
+        public void CanCaptureLastParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            Assert.Equal("World!", capture.Last);
+        }
+
+        [Fact]
+        public void CanCaptureSingleParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+
+            Assert.Equal("Hello!", capture.Single);
+        }
+
+        [Fact]
+        public void ThrowsIfSingleIsCalledWithNoParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            Assert.Throws<InvalidOperationException>(() => capture.Single);
+        }
+
+        [Fact]
+        public void ThrowsIfSingleIsCalledWithManyParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            Assert.Throws<InvalidOperationException>(() => capture.Single);
+        }
+
+        [Fact]
+        public void CanRunCallbackOnParameter()
+        {
+            var capture = new MockCapture<FooParam>();
+            capture.RunOnCapture(x => x.Value = 42);
```

Comment:
The test only uses the capture for the callback because it is a unit test. This feature is supposed to be used when the developer needs to capture values **and** to run a callback. In this case the `RunOnCapture` is quite usefull and more simple to use for methods with many parameters.